app/src/main/java/com/randomappsinc/studentpicker/grouping/GroupMakingSettingsViewHolder.java

```diff
@@ -75,35 +82,64 @@ private void applyNumberOfGroups() {
         numGroups.clearFocus();
     }
 
+    @OnCheckedChanged(R.id.autocomplete)
+    void onAutocompleteSelected(boolean checked) {
+        if (checked) {
+            if (isInputValid()) {
+                namesPerGroup.setText(getMatchingNumber(numberOfGroupsInput));
+            }
+        }
+    }
+
     @OnTextChanged(value = R.id.num_of_names_per_group, callback = OnTextChanged.Callback.AFTER_TEXT_CHANGED)
     void afterNamesPerGroupChanged() {
-        String namesPerGroupInput = namesPerGroup.getText().toString().trim();
-        boolean isInputValid = isInputValid(namesPerGroupInput);
-        if (namesPerGroup.isFocused() && isInputValid) {
+        namesPerGroupInput = namesPerGroup.getText().toString().trim();
+        boolean isInputValid = isInputValid();
```

Comment:
Shouldn't we compute this after the autocomplete potentially happens? Like if 1 input is empty, this will evaluate to false, but then the autocomplete will fill in the other input (making it valid). But since this was calculated before the autocomplete change, the positive button gets turned off.

Reply:
If the autocomplete change the value the isInputValid will be true since the autocomplete will fire the OnTextChanged and all the input will be valid.